File: expanded-form/expandedForm.js
const expandedForm = (number) => {
    const toString = number.toString();
    let output = '';
    for(let i = 0; i < toString.length - 1; i++) {
        if(toString[i] !== '0'){
            let length = toString.length - 1 - i;
            if(length > 1){
                output += toString[i] + '0'.repeat(length) + ' + ';
            } else {
                output += toString[i] + '0'  + ' + ';
            };
        };
    };

    if(toString[toString.length - 1] === '0'){
        return output.slice(0, output.length - 3);
    } else {
        return output + toString[toString.length - 1];
    };
};

module.exports = { expandedForm };

//https://www.codewars.com/kata/5842df8ccbd22792a4000245/train/javascript

File: python-venv/expanded_form/kata_function/expanded_form.py
def expanded_form(integer):
    if len(str(integer)) < 2:
        return str(integer)
    output = ''

    for i in range(len(str(integer)) - 1):
        if str(integer)[i] != '0':
            length = len(str(integer)) - 1 - i
            if length > 1:
                zeros = '0' * length
                output += str(integer)[i] + zeros + ' + '
            else:
                output += str(integer)[i] + '0' + ' + '

    if str(integer)[len(str(integer)) - 1] == '0':
        return output[0:len(output) - 3]
    else:
        return output + str(integer)[len(str(integer)) - 1]
    
#https://www.codewars.com/kata/5842df8ccbd22792a4000245/train/python